src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,177 @@
+    {
+        if ('' === $delimiter) {
+            throw new InvalidArgumentException('Passing an empty delimiter is not supported by this method. Use getIterator() method instead.');
+        }
+
+        $chunks = array();
+        foreach (preg_split('/'.preg_quote($delimiter).'/u', $this->string, $limit ?: -1) as $i => $string) {
+            $chunks[$i] = $chunk = clone $this;
+            $chunk->string = $string;
+        }
+
+        return $chunks;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toLowerCase(): self
+    {
+        $result = clone $this;
+        $result->string = mb_strtolower($this->string, 'UTF-8');
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toUpperCase(): self
+    {
+        $result = clone $this;
+        $result->string = mb_strtoupper($this->string, 'UTF-8');
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function substr(int $start = 0, int $length = null): self
+    {
+        $result = clone $this;
+        $result->string = mb_substr($this->string, $start, $length, 'UTF-8');
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trim(string $charsList = null): self
+    {
+        $charsList = $charsList ? preg_quote($charsList) : '[:space:]';
+
+        $result = clone $this;
+        $result->string = preg_replace("/^[$charsList]+|[$charsList]+\$/u", '', $this->string);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trimLeft(string $charsList = null): self
+    {
+        $charsList = $charsList ? preg_quote($charsList) : '[:space:]';
+
+        $result = clone $this;
+        $result->string = preg_replace("/^[$charsList]+/u", '', $this->string);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trimRight(string $charsList = null): self
+    {
+        $charsList = $charsList ? preg_quote($charsList) : '[:space:]';
+
+        $result = clone $this;
+        $result->string = preg_replace("/[$charsList]+\$/u", '', $this->string);
+
+        return $result;
+    }
+
+    public function startsWith(string $prefix, int $offset = 0): bool
+    {
+        return $offset === mb_strpos($this->string, $prefix, $offset, 'UTF-8');
+    }
+
+    public function endsWith(string $suffix): bool
+    {
+        $suffix = (string) $suffix;
+
+        return mb_substr($this->string, -mb_strlen($suffix), null, 'UTF-8') === $suffix;
+    }
+
+    public function getIterator(int $maxChunkLength = 1): \Generator
+    {
+        if ($maxChunkLength < 1) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ($maxChunkLength > 65535) {
+            $maxChunkLength = 65535;
+        }
+
+        if ('' === $this->string) {
+            yield clone $this;
```

Comment:
just "return"? str_split return `array('')` with empty strings, but I don't get why this is useull at all.
WDYT? (same in the other "getIterator")